### PR TITLE
[diffusion] feat: cache encoder results for default negative prompt

### PR DESCRIPTION
**Pull request by mickqian**
## Summary

- Add a one-entry cache for deterministic negative prompt text encoder outputs in `TextEncodingStage`.
- Do not read from or populate the cache for warmup requests, so warmup does not precompute speculative data for later real requests.
- Keep the cache key to the values that affect this encode path: pipeline class, encoder set, negative prompt text, prompt template, and max sequence length.

For LTX-2.3, the cached default negative embedding is `[1, 1024, 188160]` bf16, about 367 MiB. `1024` is the token sequence length, not a layer count. The cache is capped at one entry, so this is also the expected upper-bound tensor overhead for this path.

## Performance

Single H100, `CUDA_VISIBLE_DEVICES=2`, `--pipeline-class-name LTX2TwoStagePipeline`, `--ltx2-two-stage-device-mode snapshot`, two real 24-frame requests in the same server after warmup. The first request fills the cache; the second request has the same negative prompt and hits the cache.

| Metric | Request 1 | Request 2 | Delta |
| --- | ---: | ---: | ---: |
| `TextEncodingStage` | 1035.5 ms | 471.9 ms | -563.6 ms / -54.4% |
| summed logged pipeline stages | 7737.8 ms | 7071.0 ms | -666.8 ms / -8.6% |
| `LTX2AVDenoisingStage` | 5413.0 ms | 5342.5 ms | -70.5 ms |
| `LTX2RefinementStage` | 388.3 ms | 382.6 ms | -5.7 ms |
| `LTX2AVDecodingStage` | 787.2 ms | 774.5 ms | -12.7 ms |

## Precision

Compared the request-1 cache-fill output against the request-2 cache-hit output from the same server, same prompt, same seed:

| Frame | max abs diff | mean abs diff |
| --- | ---: | ---: |
| first | 0 | 0.000000 |
| middle | 0 | 0.000000 |
| last | 0 | 0.000000 |

## Validation

- `python -m py_compile python/sglang/multimodal_gen/runtime/pipelines_core/stages/text_encoding.py`
- `git diff --check origin/main...HEAD`
- Remote H100 two-request cache run with `seconds=1`, same prompt/seed/negative prompt
- Remote H100 frame comparison for the two outputs, all compared frames exact (`max_abs_diff=0`)
- Added unit coverage for cache-key misses on `prompt_template` / `max_sequence_length`, and for warmup not populating the cache

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/text_encoding.py [hunk 1]
```diff
@@ -123,12 +124,38 @@ def forward(
...
             assert isinstance(batch.negative_prompt, str)
-            neg_embeds_list, neg_masks_list, neg_pooler_embeds_list = self.encode_text(
+            negative_cache_key = (
+                server_args.pipeline_class_name,
+                tuple(all_indices),
                 batch.negative_prompt,
-                server_args,
-                encoder_index=all_indices,
-                return_attention_mask=True,
+                batch.max_sequence_length,
```
[the remark this answers is not in the document]
Comment (mickqian) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/text_encoding.py]: Fixed in `63d138b7b`. I removed `batch.max_sequence_length` from the key because this generic `TextEncodingStage.encode_text()` call does not pass or consume it in the tokenizer path, so keeping it only caused unnecessary misses.

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/text_encoding.py [hunk 2]
```diff
@@ -123,12 +124,38 @@ def forward(
...
+                # Negative prompts are often a long model default that warmup
+                # and later requests share. For the same pipeline, encoders,
+                # prompt text, and max length, encoding is deterministic, so the
+                # cached tensors are exactly the result of calling encode_text
+                # again.
+                self._negative_text_cache[negative_cache_key] = (
+                    tuple(neg_embeds_list),
+                    tuple(neg_masks_list),
+                    tuple(neg_pooler_embeds_list),
+                )
```
[the remark this answers is not in the document]
Comment (mickqian) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/text_encoding.py]: I checked the LTX-2.3 downstream path: `LTX2TextConnectorStage` reads these tensors and writes new connector outputs back to the batch, and denoising only reads/cats them. There is no in-place mutation of the cached text encoder outputs in this path. I kept references instead of cloning because cloning the default LTX-2.3 negative embedding would add another ~367 MiB copy and reduce the cache benefit. Verified with two same-server requests: cache-fill output vs cache-hit output matched exactly on first/middle/last frames (`max_abs_diff=0`).
